ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Iterator.hs

```diff
@@ -141,16 +144,19 @@ import qualified Ouroboros.Storage.ChainDB.Impl.VolDB as VolDB
 --   read from disk.
 --
 -- * When blocks have to be streamed both from the ImmutableDB and the
```

Comment:
IIUC, your change here clarifies that the hash of the tip of the ImmutableDB is already in memory, so we can avoid a disk-read for that block. But that got me thinking:

This whole `Bounds checking` section seems a little loose regarding the distinction between _the bounds of the requested interval_ versus _the bounds of the interval's (not necessarily proper) prefix that is to be streamed from the ImmutableDB_.

Part of the subtlety in the buggy case that #231 revealed is that the _computed path_ involves both the Immutable and Volatile DBs, but _the streaming_ could be all from the Immutable DB. Moreover, the (iteration-creation time) Immutable DB tip won't be involved. So this cost discussion might need to be adjusted once the bug is fixed?

Reply:
I have rewritten this section to clarify this.

Also see my fix for the bug which avoids streaming from both when we can stream from just the ImmutableDB.